lib/gollum/views/helpers/locale_helpers.rb

```diff
@@ -0,0 +1,81 @@
+module Precious
+  module Views
+    module LocaleHelpers
+      NO_METHOD_MESSAGE = 'Argument must be a view method'
+
+      # Returns all I18n translation strings for the current view class.
+      # This method support YAML arguments. For example:
+      #
+      #     last_edited: This content was last edited at %{date}.
+      #
+      # Where the `date` argument must be a method available on the current
+      # class.
+      #
+      # Use this interface within Mustache templates to render any user
+      # interface strings in the current locale. For example:
+      #
+      #     {{ t.last_edited }}
+      #
+      def t
+        autofill I18n.t(locale_klass_name)
+      end
+
+      private
+
+      # Recursively looks up I18n translation values and autofills any YAML
+      # arguments with the return value of the current class's matching method.
+      #
+      # When a translation value with an argument has no matching method, we
+      # then return that value transformed to include the `no_method_message`
+      #
+      def autofill(yaml)
+        yaml.map { |i18n_key, i18n_value|
+          if i18n_value.is_a? Hash
+            [i18n_key, autofill(i18n_value)]
+          elsif has_arguments?(i18n_value)
+            fill_argument_content(i18n_key, i18n_value)
+          else
+            [i18n_key, i18n_value]
+          end
+        }.to_h
+      end
+
+      def fill_argument_content(i18n_key, i18n_value)
+        i18n_value.gsub!(/\%\{[\w]+\}/) do |argument|
```

Comment:
Since you're using the same regex again in `#has_arguments?`, you could store it in an appropriately named constant. But the regex is very readable, I'm fine with it this way!

Reply:
I decided to add a constant for this regexp. Thanks for the feedback!